Documentation/common-problems.md

```diff
@@ -23,25 +26,149 @@ One of the first things that should be done is to start the [rook-tools pod](./t
+
+## Investigation Details
+If you see some of the symptoms above, it's because the requested Rook storage for your pod is not being created and mounted successfully.
+To confirm this issue, you can run the following commands and you should see similar output (note that some of it has been omitted for brevity):
+```console
+> kubectl get pod
+NAME                              READY     STATUS              RESTARTS   AGE
+wordpress-mysql-918363043-50pjr   0/1       ContainerCreating   0          1h
+
+> kubectl describe pod wordpress-mysql-918363043-50pjr
+...
+Events:
+  FirstSeen	LastSeen	Count	From			SubObjectPath	Type		Reason			Message
+  ---------	--------	-----	----			-------------	--------	------			-------
+  1h		1h		3	default-scheduler			Warning		FailedScheduling	PersistentVolumeClaim is not bound: "mysql-pv-claim" (repeated 2 times)
+  1h		35s		36	kubelet, 172.17.8.101			Warning		FailedMount		Unable to mount volumes for pod "wordpress-mysql-918363043-50pjr_default(08d14e75-bd99-11e7-bc4c-001c428b9fc8)": timeout expired waiting for volumes to attach/mount for pod "default"/"wordpress-mysql-918363043-50pjr". list of unattached/unmounted volumes=[mysql-persistent-storage]
+  1h		35s		36	kubelet, 172.17.8.101			Warning		FailedSync		Error syncing pod
+```
+
+To troubleshoot this, let's walk through the volume provisioning steps in order to confirm where the failure is happening.
+
+### Rook Agent Deployment
+The `rook-agent` pods are responsible for mapping and mounting the volume from the cluster onto the node that your pod will be running on.
+If the `rook-agent` pod is not running then it cannot perform this function.
+Below is an example of the `rook-agent` pods failing to get to the `Running` status because they are in a `CrashLoopBackOff` status:
+```console
+> kubectl -n rook-system get pod
+NAME                             READY     STATUS             RESTARTS   AGE
+rook-agent-ct5pj                 0/1       CrashLoopBackOff   16         59m
+rook-agent-zb6n9                 0/1       CrashLoopBackOff   16         59m
+rook-operator-2203999069-pmhzn   1/1       Running            0          59m
+```
+If you see this occurring, you can get more details about why the `rook-agent` pods are continuing to crash with the following command and its sample output:
+```console
+> kubectl -n rook-system get pod -l app=rook-agent -o jsonpath='{range .items[*]}{.metadata.name}{"\t"}{.status.containerStatuses[0].lastState.terminated.message}{"\n"}{end}'
+rook-agent-ct5pj	mkdir /usr/libexec/kubernetes: read-only file system
+rook-agent-zb6n9	mkdir /usr/libexec/kubernetes: read-only file system
+```
+From the output above, we can see that the agents were not able to bind mount to `/usr/libexec/kubernetes` on the host they are scheduled to run on.
+For some environments, this default path is read-only and therefore a better path must be provided to the agents.
+
+First, clean up the agents deployment with:
```

Comment:
typo: agent or agents' (possessive?)

Reply:
`agent deployment` sounds good to me here